tests/extensions/default/analyzers/Microsoft.DotNet.UpgradeAssistant.Extensions.Default.Analyzers.Test/WebTypeReplacementsTests.cs

```diff
@@ -0,0 +1,60 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Xunit;
+
+namespace Microsoft.DotNet.UpgradeAssistant.Extensions.Default.Analyzers.Test
+{
+    public static class WebTypeReplacementsTests
+    {
+        private static readonly Dictionary<string, string> ExpectedReplacements = new()
+        {
+            { "System.Web.Http.ApiController", "Microsoft.AspNetCore.Mvc.ControllerBase" },
+            { "System.Web.Mvc.Controller", "Microsoft.AspNetCore.Mvc.Controller" },
+            { "System.Web.Mvc.ResultExecutingContext", "Microsoft.AspNetCore.Mvc.Filters.ResultExecutingContext" },
+            { "System.Web.Mvc.ResultExecutedContext", "Microsoft.AspNetCore.Mvc.Filters.ResultExecutedContext" },
+            { "System.Web.Mvc.IResultFilter", "Microsoft.AspNetCore.Mvc.Filters.IResultFilter" },
+            { "System.Web.Mvc.ActionFilterAttribute", "Microsoft.AspNetCore.Mvc.Filters.ActionFilterAttribute" },
+            { "System.Web.Mvc.ActionExecutingContext", "Microsoft.AspNetCore.Mvc.Filters.ActionExecutingContext" },
+            { "System.Web.Mvc.ActionExecutedContext", "Microsoft.AspNetCore.Mvc.Filters.ActionExecutedContext" },
+            { "System.Web.Mvc.IActionFilter", "Microsoft.AspNetCore.Mvc.Filters.IActionFilter" },
+            { "System.Web.WebPages.HelperResult", "Microsoft.AspNetCore.Mvc.Razor.HelperResult" },
+            { "System.Web.HtmlString", "Microsoft.AspNetCore.Html.HtmlString" },
+            { "System.Web.IHtmlString", "Microsoft.AspNetCore.Html.HtmlString" },
+            { "System.Web.Mvc.MvcHtmlString", "Microsoft.AspNetCore.Html.HtmlString" },
+            { "System.Web.Mvc.ActionResult", "Microsoft.AspNetCore.Mvc.ActionResult" },
+            { "System.Web.Mvc.ContentResult", "Microsoft.AspNetCore.Mvc.ContentResult" },
+            { "System.Web.Mvc.FileResult", "Microsoft.AspNetCore.Mvc.FileResult" },
+            { "System.Web.Mvc.HttpNotFoundResult", "Microsoft.AspNetCore.Mvc.NotFoundResult" },
+            { "System.Web.Mvc.HttpStatusCodeResult", "Microsoft.AspNetCore.Mvc.StatusCodeResult" },
+            { "System.Web.Mvc.HttpUnauthorizedResult", "Microsoft.AspNetCore.Mvc.UnauthorizedResult" },
+            { "System.Web.Mvc.RedirectResult", "Microsoft.AspNetCore.Mvc.RedirectResult" },
+            { "System.Web.Mvc.PartialViewResult", "Microsoft.AspNetCore.Mvc.PartialViewResult" },
+            { "System.Web.Mvc.ViewResult", "Microsoft.AspNetCore.Mvc.ViewResult" },
+        };
+
+        [Fact]
+        public static void WebTypeReplacementContentTests()
+        {
+            // Arrange
+            var replacementsText = new WebTypeReplacements();
```

Comment:
I'm not sure we want to map them with a custom class like this. It's pretty hard-coded

Reply:
loading data from the class is going to be much faster than loading the data from disk. These examples are built to run in VS so that optimization seemed like part of the design decision.

Agree that extending this would be more flexible if we could read from disk, but not sure we'd want that feature running in VS.

Reply:
I've re-worked this to load type mappings from configuration files so that they work better in the standalone analyzer package case. It probably is less performant, but I don't think it will matter. UA consumers can still supply AdditionalText files however they like (either by deriving from the class or using the `AdditionalFileText` I've added to load from disk).